testing/test_float4.cpp

```diff
@@ -2,26 +2,23 @@
 #include <tack/float4.h>
 #include <array>
 
-struct test_float4_abs_t : public test_t {
+TEST(float4, abs)
+{
+    using namespace tack;
+    const float4 a(-1, 2, -3, 4);
+    const float4 b = abs(a);
+    std::array<float, 4> out;
+    b.store(out.data());
+    ASSERT(out[0] == 1.f && out[1] == 2.f && out[2] == 3.f && out[3] == 4.f);
+}
 
-    test_float4_abs_t()
-        : test_t("test_float4_abs")
-    {
-        register_test(this);
-    }
-
-    virtual bool run() override
-    {
-        using namespace tack;
-        const float4 a(-1, 2, -3, 4);
-        const float4 b = abs(a);
-        std::array<float, 4> out;
-        b.store(out.data());
-        pass_ &= out[0] == 1.f && out[1] == 2.f && out[2] == 3.f && out[3] == 4.f;
-        return passed();
-    }
-};
-
-namespace {
-test_float4_abs_t test_1;
-} // namespace {}
+TEST(float4, equal)
```

Comment:
This is quite nice, its a lot more concise and removing the boiler plate is good.